docs/en/basic_concepts/structures.md

```diff
@@ -1,3 +1,281 @@
+
+```python
+# Step 1: DBHead, input the initial image img and data_samples from the data pipeline, and return the output of the network part of the model
+def forward(self, img: Tensor,
+        data_samples: Optional[List[TextDetDataSample]] = None
+    ) -> Tuple[Tensor, Tensor, Tensor]:
+
+# (Training) Step 2: DBModuleLoss, input the output of the DBNet network part and data_samples
+def forward(self, preds: Tuple[Tensor, Tensor, Tensor],
+                data_samples: Sequence[TextDetDataSample]) -> Dict:
+
+# (Test) Step 2: DBPostProcessor
+def get_text_instances(self, pred_results: Tuple[Tensor, Tensor, Tensor],
+                           data_sample: TextDetDataSample
+                           ) -> TextDetDataSample:
+```
+
+As you can see from the above example, [`TextDetDataSample`](mmocr.structures.TextRecogDataSample) is used throughout the training and testing process of the detection model, and it encapsulates the data needed for the whole process of the text detection task.
+
+### Text Recognition Task Data Abstraction TextRecogDataSample
+
+[`TextRecogDataSample`](mmocr.structures.TextRecogDataSample) is used to encapsulate the data for the text recognition task. It has two properties, `gt_text` and `pred_text` , which are used to store prediction results and annotation information respectively.
+
+|           |                           |                        |
+| --------- | ------------------------- | ---------------------- |
+| Field     | Type                      | Description            |
+| gt_text   | [`LabelData`](#labeldata) | Label information.     |
+| pred_text | [`LabelData`](#labeldata) | The prediction result. |
+
+The following sample code demonstrates the use of [`TextRecogDataSample`](mmocr.structures.TextRecogDataSample).
+
+```python
+import torch
+from mmengine.data import TextRecogDataSample
+
+data_sample = TextRecogDataSample()
+# Define the ground truth data
+img_meta = dict(img_shape=(800, 1196, 3), pad_shape=(800, 1216, 3))
+gt_text = LabelData(metainfo=img_meta)
+gt_text.item = 'mmocr'
+data_sample.gt_text = gt_text
+
+# Define the prediction data
+pred_text = LabelData(metainfo=img_meta)
+pred_text.item = 'mmocr'
+data_sample.pred_text = pred_text
+```
+
+The fields of `Labelata` that will be used are:
+
+|       |                     |                                                                                           |
+| ----- | ------------------- | ----------------------------------------------------------------------------------------- |
+| Field | Type                | Description                                                                               |
+| item  | `list[str]`         | The text corresponding to the instance, of length (N, ), for end-to-end OCR tasks and KIE |
+| score | `torch.FloatTensor` | Confidence of the text prediction, of length (N, ), for the end-to-end OCR task           |
```

Comment:
Missing `indexes` and `padded_indexes`
https://github.com/open-mmlab/mmocr/blob/8483b176b2d268303495666791a5a0e1535777b8/mmocr/models/textrecog/module_losses/base.py#L129-L130